extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -120,6 +121,37 @@ export class AmpAnalytics extends AMP.BaseElement {
           'config. No analytics data will be sent.');
       return Promise.resolve();
     }
+    if (this.config_['extraUrlParams'] &&
+        this.config_['extraUrlParamsReplaceMap']) {
+      // If the config includes a extraUrlParamsReplaceMap, apply it as a set
+      // of params to String.replace to allow aliasing of the keys in
+      // extraUrlParams.
+      let count = 0;
+      for (let replaceMapKey in this.config_['extraUrlParamsReplaceMap']) {
+        if (++count > MAX_REPLACES) {
+          console./*OK*/error(this.getName_(),
+           "More than " + MAX_REPLACES.toString() +
+           " extraUrlParamsReplaceMap rules aren't allowed; Skipping the rest"
+          );
+          break;
+        }
+
+        // only allow word chars
+        replaceMapKey = replaceMapKey.replace(/(\W)/g, '\$1');
```

Comment:
early iterations treated the key as a regex.  It's now just a string, though.    I'll remove the line as it's irrelevant now.